packages/core/strapi/lib/types/core/strapi/index.d.ts

```diff
@@ -5,6 +5,28 @@ import type { StringMap } from './utils';
 import type { GenericController } from '../../../core-api/controller'
 import type { GenericService } from '../../../core-api/service'
 
+// TODO move custom fields types to a separate file
```

Comment:
We'll discuss with JS in a couple weeks where would be the best place for this type